### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by McKucia**

### CODE REVIEW

File: module2/homework/addEven/addEven.hpp [hunk 1]
```diff
@@ -2,7 +2,9 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+        int sum=0;
+    for(const auto& el : numbers)
+        if( el%2 == 0)
```
Comment (piezol) [inline module2/homework/addEven/addEven.hpp:7]:
```suggestion
        if(el%2 == 0)
```

File: module2/homework/generateSequence/generateSequence.hpp [hunk 2]
```diff
@@ -2,6 +2,12 @@
 #include <vector>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+    if(count <=0)
+    	return {};
+    std::vector<int> vec; 
+    vec.reserve(count); 
+    for(size_t i=1; i<=count; i++)
```
Comment (piezol) [inline module2/homework/generateSequence/generateSequence.hpp:9]:
```suggestion
    for(size_t i=1; i <= count; i++)
```

File: module2/homework/nwdNww/nwdNww.hpp [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+        return NWD(rhs, lhs % rhs);
+    else
+        return std::abs(lhs);
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    if(lhs != 0 && rhs != 0)
+    	return std::abs((lhs * rhs)/NWD(lhs, rhs));
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:13]: I'm copying my comment from different PR:
There are trivial cases that your code won't work for, ie NWW(1e8, 1e8).
That is due to lhs*rhs overflowing int.
You can fix it by casting lhs to unsigned long long in the multiplication.

There are some nontrivial cases that will still fail, ie NWW(1e8-1, 1e8+1), but they fail for the std:lcm(int, int) as well, so it's ok.

File: module2/homework/generateSequence/generateSequence.hpp [hunk 4]
```diff
@@ -2,6 +2,12 @@
...
+        vec.push_back(step*i);
```
Comment (NeoBauer) [inline module2/homework/generateSequence/generateSequence.hpp:10]: There was a suggestion somewhere in our PR for Homework2, that it is better to add, than multiply that value. Possibility to get faster code ;).

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635], "points": [2, 4, 2, 6, 2, 5, 2, 5] }

**Pull request closed without merging**